src/lib/sap_ha/system/network.rb

```diff
@@ -41,15 +41,19 @@ def interfaces
       # Get local machine's IPv4 addresses excluding the loopback iface
       def ip_addresses
         interfaces = Socket.getifaddrs.select do |iface|
-          iface.addr.ipv4? && !iface.addr.ipv4_loopback?
+	  if !iface.addr.nil?
```

Comment:
You can simplify it to a single statement
```suggestion
	  !iface.addr.nil? && iface.addr.ipv4? && !iface.addr.ipv4_loopback?
```

The problem with `if ` is when the condition is not fullfiled it not true then it the result is evaluated to `nil`. In this case it behaves as `false` so it works fine, but it still does not look nice and might be confusing.

Reply:
Thanks a lot!